src/SearchWrapper.cpp

```diff
@@ -190,21 +180,21 @@ searchWrapper(Rcpp::CharacterVector & dbpath, Rcpp::List & enquireList, Rcpp::Li
 	    output[str] = rowVec;
 	}
 
-	bool displaySpies = true;
-	if (enquireList.containsElementNamed("viewSpies")) displaySpies = Rcpp::as<bool>(enquireList["viewSpies"]);
-
-	if (displaySpies) {
-	    // Fetch and display the spy values
-	    for (std::vector<Xapian::ValueCountMatchSpy *>::iterator itr = spy.begin(); itr != spy.end(); ++itr) {
-		Rcpp::Rcout << "new spy" << std::endl;
-		Xapian::ValueCountMatchSpy * aSpy = *itr;
-		for (Xapian::TermIterator facet = aSpy->values_begin(); facet != aSpy->values_end(); ++facet) {
-		    Rcpp::Rcout << "Facet: " << *facet << "; count: " << facet.get_termfreq() << std::endl;
-		}
+	if (returnSpy) {
+	    Rcpp::List outputSpy;
+	    std::string aStr;
+	    for (Xapian::TermIterator facet = spy->values_begin(); facet != spy->values_end(); ++facet) {
+		aStr = *facet + "; count: ";
+		int termFreq = facet.get_termfreq();
+		stringstream convert;
+		convert << termFreq;
+		aStr += convert.str();
+		outputSpy.push_back(aStr);
 	    }
-	}
+	    return outputSpy;
+	} else
+	    return convertToDataFrame(output);
 
-	return convertToDataFrame(output);
```

Comment:
Done :)

On Sun, Jul 24, 2016 at 7:27 PM, James Aylett <notifications@github.com>
wrote:

> In src/SearchWrapper.cpp
> <https://github.com/amandaJayanetti/RXapian/pull/11#discussion_r71989034>:
>
> >
> > -	return convertToDataFrame(output);
>
> Is it possible to get access to both the spies and the output? That would
> be fairly common usage, I think.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/amandaJayanetti/RXapian/pull/11/files/62f7315d2db8d1b60505a92c6c372e0e2a730954..42b7c64c98f0dcabf5e41955fc7d6ed1b046737a#r71989034>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH_jNsi6a50iBnc1OSpOuJdFJEGu-5qcks5qY29WgaJpZM4JDzJo>
> .
>